manifests/server/root_password.pp

```diff
@@ -16,26 +16,19 @@
   }
 
   $options = $mysql::server::_options
-  $secret_file = $mysql::server::install_secret_file
   $login_file = $mysql::server::login_file
 
-  # Validate that the given secret_file value is a valid path
-  if $secret_file !~ /^(?:\/[A-Za-z0-9._-]+)+?$/ {
-    $message = "Invalid value provided for parameter install_secret_file. '${secret_file}' is not a valid path."
-    fail($message)
-  }
-
   # New installations of MySQL will configure a default random password for the root user
   # with an expiration. No actions can be performed until this password is changed. The
   # below exec will remove this default password. If the user has supplied a root
   # password it will be set further down with the mysql_user resource.
   $rm_pass_cmd = join([
-      "mysqladmin -u root --password=\$(grep -o '[^ ]\\+\$' ${secret_file}) password ''",
-      "rm -f ${secret_file}",
+      "mysqladmin -u root --password=\$(grep -o '[^ ]\\+\$' /.mysql_secret) password ''",
+      'rm -f /.mysql_secret',
   ], ' && ')
```

Comment:
This should be chained with a `;` not `&&`. If the `mysqladmin` command fails with the current command chaining it won't delete the `.mysql_secret` file

Reply:
I think that's desired though. If we remove the file without setting the password, then we've lost it.

Reply:
🤔 we don't want to leave a cleartext file sitting on the hard drive though

Reply:
Let's look at the workflow and fragilities. This is how I understand it, please correct me if I'm wrong.

1. the mysql install runs and generates a randomized password
2. this is saved in plaintext on disk. It allows you only the ability to set your own password--you can't use it for normal operations.
3. the user is expected to specify a new password and delete the temporary password file, which is what this command string is doing.

If step 3 fails, I can only think of a handful of causes:
1. mysql installation failed or the service didn't start. Manual intervention is required, and the plaintext password may or may not be valid.
2. race condition -- the install succeeded, but an unknown actor already changed the root password. The plaintext password is invalid, but this should probably trigger an investigation.
3. the process that generated the install password and file on disk failed. The plaintext password is invalid.

Are there other scenarios?

Reply:
yes, assume that something else went wrong with the `mysqladmin` password change command, in this case the password was not changed or removed, and it's still sitting on the disk, right? The safer option here is to just delete it regardless of success or failure

Reply:
I'd still call that scenario 1 and the proper response imho is to fail and ensure manual intervention. I'm not sure if I care about the password file existing. It could make troubleshooting easier, but I suspect that most users won't bother--they'll either burn the node or remove the installation and try again.

My concern with chaining with `;` is that it will make that exec always succeed and so it won't bubble up errors.

Reply:
how about we do it like this

```shell
 mysqladmin -u root --password=\$(grep -o '[^ ]\\+\$' /.mysql_secret) password \
&& (rm -f  /.mysql_secret; exit 0) \
|| (rm -f /.mysql_secret; exit 1)
 ```
 Now we'll always delete the secret, and we'll have the error if something went wrong?

Reply:
Ha! I looked at the original code again and it wasn't surfacing errors to start with!